app/concepts/cinema_halls/repository.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module CinemaHalls
+  class Repository
+    attr_reader :adapter
+
+    def initialize(adapter: CinemaHall)
+      @adapter = adapter
+    end
+
+    def all
+      adapter.all
+    end
+
+    def find(id)
+      adapter.find(id)
+    end
+
+    def create(params)
+      adapter.create(params)
+    end
+
+    def update(id, params)
+      cinema_hall = adapter.find(id)
+      cinema_hall.update(params)
+      cinema_hall
+    end
+
+    def delete(id)
+      adapter.find(id).delete
+    end
```

Comment:
Now, when you have similar behavior for more than one repository, you can try to create BaseRepository class. Of course it should be moved to next PR and consider it like an additional thing to do :)